msgraph/identity_providers_test.go

```diff
@@ -0,0 +1,125 @@
+		randomString: rs,
+	}
+	c.client = msgraph.NewIdentityProvidersClient(c.connection.AuthConfig.TenantID)
+	c.client.BaseClient.Authorizer = c.connection.Authorizer
+
+	testIdentityProvidersClient_ListAvailableProviderTypes(t, c)
+
+	identityProvider := testIdentityProvidersClient_Create(t, c, msgraph.IdentityProvider{
+		ODataType:    utils.StringPtr("microsoft.graph.socialIdentityProvider"),
+		Name:         utils.StringPtr("Login with Google"),
+		Type:         utils.StringPtr("Google"),
+		ClientId:     utils.StringPtr("56433757-cadd-4135-8431-2c9e3fd68ae8"),
+		ClientSecret: utils.StringPtr("000000000000"),
+	})
+
+	testIdentityProvidersClient_Get(t, c, *identityProvider.ID)
+
+	patchIdentityProvider := &msgraph.IdentityProvider{}
+	patchIdentityProvider.ODataType = identityProvider.ODataType
+	patchIdentityProvider.ID = identityProvider.ID
+	patchIdentityProvider.ClientSecret = utils.StringPtr("1111111111111")
+	testIdentityProvidersClient_Update(t, c, *patchIdentityProvider)
+
+	testIdentityProvidersClient_List(t, c)
+
+	testIdentityProvidersClient_Delete(t, c, *identityProvider.ID)
+}
+
+func testIdentityProvidersClient_Create(t *testing.T, c IdentityProvidersClientTest, p msgraph.IdentityProvider) (provider *msgraph.IdentityProvider) {
+	provider, status, err := c.client.Create(c.connection.Context, p)
+	if err != nil {
+		t.Fatalf("IdentityProvidersClient.Create(): %v", err)
+	}
+	if status < 200 || status >= 300 {
+		t.Fatalf("IdentityProvidersClient.Create(): invalid status: %d", status)
+	}
+	if provider == nil {
+		t.Fatal("IdentityProvidersClient.Create(): provider was nil")
+	}
+	if provider.ID == nil {
+		t.Fatal("IdentityProvidersClient.Create(): provider.ID was nil")
+	}
+	return
+}
+
+func testIdentityProvidersClient_Update(t *testing.T, c IdentityProvidersClientTest, p msgraph.IdentityProvider) {
+	status, err := c.client.Update(c.connection.Context, p)
+	if err != nil {
+		t.Fatalf("IdentityProvidersClient.Update(): %v", err)
+	}
+	if status < 200 || status >= 300 {
+		t.Fatalf("IdentityProvidersClient.Update(): invalid status: %d", status)
+	}
+}
+
+func testIdentityProvidersClient_List(t *testing.T, c IdentityProvidersClientTest) (providers *[]msgraph.IdentityProvider) {
+	providers, _, err := c.client.List(c.connection.Context)
+	if err != nil {
+		t.Fatalf("IdentityProvidersClient.List(): %v", err)
+	}
+	if providers == nil {
+		t.Fatal("IdentityProvidersClient.List(): providers was nil")
+	}
+	return
+}
+
+func testIdentityProvidersClient_Get(t *testing.T, c IdentityProvidersClientTest, id string) (provider *msgraph.IdentityProvider) {
+	provider, status, err := c.client.Get(c.connection.Context, id)
+	if err != nil {
+		t.Fatalf("IdentityProvidersClient.Get(): %v", err)
+	}
+	if status < 200 || status >= 300 {
+		t.Fatalf("IdentityProvidersClient.Get(): invalid status: %d", status)
+	}
+	if provider == nil {
+		t.Fatal("IdentityProvidersClient.Get(): provider was nil")
+	}
+	return
```

Comment:
nil check here to avoid later crash (this needs adding in some other client tests too)

```suggestion
	if provider.ID == nil {
		t.Fatal("IdentityProvidersClient.Create(): provider.ID was nil")
	}
		return
```